design-system/react/src/components/InputAmount/InputAmount.tsx

```diff
@@ -93,31 +136,59 @@ export const InputAmount: InputAmountComponent = ({
       {initialBalance && (
         <Input.ElementRight css={styles.elementRight}>
           <Box css={styles.balanceActions}>
-            {!hiddenMaxButton && (
-              <Flex align="center" justify="end">
-                <Button
-                  aria-label="Max"
-                  size="sm"
-                  variant="ghost"
-                  onPress={handleSetBalance}
-                  css={styles.maxButton}
-                >
-                  Max
-                </Button>
-              </Flex>
-            )}
-            {!hiddenBalance && (
-              <Flex
-                as="div"
-                css={styles.balance}
-                aria-label={`Balance: ${formattedBalance}`}
-              >
-                <Box as="span">Balance: </Box>
-                <Tooltip content={formatAmount(balance)} sideOffset={-5}>
-                  <Box as="span">{formattedBalance}</Box>
-                </Tooltip>
+            <Flex align="end" direction="column">
+              {assetInfo && assetInfo.length > 0 && onClickAsset && (
+                <Flex>
+                  <Dropdown>
+                    <Dropdown.Trigger>
+                      <Button
+                        size="xs"
+                        aria-label="Coin Selector"
+                        css={{
+                          background: '$blackA12',
+                          color: '$gray9',
+                        }}
+                      >
+                        {tokenImage(assetText, assetImageUrl)}
+                        {assetText}
+                      </Button>
+                    </Dropdown.Trigger>
+                    <Dropdown.Menu
+                      onAction={(e) => handleSelectAsset(e.toString())}
+                    >
+                      {dropdownItems}
+                    </Dropdown.Menu>
+                  </Dropdown>
+                </Flex>
```

Comment:
On the fuel-ui side, this should be just a button that enables the upper application to implement the selector in a better way like a popup.